src/test/proxy/worker_proxy_factory.py

```diff
@@ -42,3 +41,35 @@ def test_retrieval(self) -> None:
                         saved_proxy, retrieved_proxy,
                         "Retrieved a different proxy for index %s in permutation %s" % (index, indices)
                     )
+
+
+class WorkerProxyDummyFactory(WorkerProxyFactoryBase):
+    """
+    Factory class for WorkerProxyDummy.
+    """
+    def _create_proxy(self, work_machine: WorkMachine) -> IWorkerProxy:
+        return WorkerProxyDummy(uid=work_machine.uid, jobs=[])
+
+
+class WorkerProxyDummyFactoryTest(AbstractWorkerProxyFactoryTest):
+    """
+    Class for testing WorkerProxyDummyFactory.
+    """
+    def _get_factory(self) -> WorkerProxyFactoryBase:
+        return WorkerProxyDummyFactory(database=None)
+
+
+class WorkerProxyFactory(WorkerProxyFactoryBase):
```

Comment:
I find the name confusing because it is the same name as the actual `WorkerProxyFactory`. Can we name it `Mock...` or `...NoCache` or something like that?